### ISSUE TITLE
Report generation doesn't take into account broken fuzzers well.

### ISSUE THREAD

**Issue by jonathanmetzman** [google/fuzzbench#482]
In https://www.fuzzbench.com/reports/2020-06-18/index.html the four fuzzers that aren't aflplusplus_qemu finish on bloaty_fuzz_target. But because all qemu trials for bloaty_fuzz_target end very early, the report stops at one hour. 
It might not be possible to come up with enough heuristics to make perfect report generation, but this behavior seems non-ideal. We used to drop fuzzers on benchmarks if too many of their trials failed. That might be a good fix. I wonder if the 80% heuristic we use now should be gotten rid of. Probably worth looking at our data and seeing how many times it helps.

**Comment by jonathanmetzman** [comment]
cc @lszekeres

**Comment by lszekeres** [comment]
Right, the question is what to show in the report when we have incomplete data, ie. when we don't have 20 trials for each fuzzer that ran for 24 hours. We could drop fuzzers for which this happens, but that way we're throwing out useful data, even if it's partial. By using using end time time before 24h in the report where most trials are still alive, we similarly drop some potentially useful (but partial) data. 

I think the best might be to display all data (even when it's incomplete), and just display warnings about the incomplete fuzzers. We  already have these warnings, and if we don't want to shorten the reported trial time, we just need to set `BENCHMARK_SAMPLE_NUM_THRESHOLD=1.0`. Wdyt?

**Comment by zchcai** [comment]
Is this relevant to #481? Can we dynamically select fuzzers to generate reports? For example, for the 2020-06-18, if we provide a way to let users select those fuzzers with more successful trials, then it will generate a report based on the minimal number successful trials among them.

> We already have these warnings, and if we don't want to shorten the reported trial time, we just need to set BENCHMARK_SAMPLE_NUM_THRESHOLD=1.0. Wdyt?

I guess this only increase the bar for putting warnings. 1.0 means warnings will put as long as there exists failed trials for one fuzzer.

> I think the best might be to display all data (even when it's incomplete)

Currently we use a minimal successful number for comparison and this makes sense. If we want to compare fuzzers with different numbers of successful trials, that is not comparable.
I guess the best way is to either give options to dynamically select fuzzers to generate, or we drop the lowest one fuzzer to generate report. I think the former can do both what we want.

**Comment by jonathanmetzman** [comment]
>I think the best might be to display all data (even when it's incomplete), and just display warnings about the incomplete fuzzers. We already have these warnings, and if we don't want to shorten the reported trial time, we just need to set BENCHMARK_SAMPLE_NUM_THRESHOLD=1.0. Wdyt?

I actually think the current behavior is good in most cases it just seems to fail when there is a small number of fuzzers and one fuzzer fails. Maybe I don't have a coherent idea of what should be done.

>Is this relevant to #481? Can we dynamically select fuzzers to generate reports? 

Mostly not. #481 could help with this a little if users could apply their own heuristics (which isn't really the purpose of #481), but this question is more about defaults.

**Comment by lszekeres** [comment]
Sorry, I remembered wrong when I originally replied to this, and I only realized after quickly submitting https://github.com/google/fuzzbench/pull/483. If we want later snapshot times (e.g., 24 hours), we actually need to lower the threshold, not increase it. The threshold represents the fraction of the trials that need to be alive at the time of snapshot, so lower values lead to later snapshot times (https://github.com/google/fuzzbench/pull/505).

Sent out https://github.com/google/fuzzbench/pull/506 lowering the threshold this time :)
